plugins/metrics/runtime/runtime.go

```diff
@@ -0,0 +1,89 @@
+package runtime
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/unit"
+	"golang.org/x/time/rate"
+
+	"github.com/open-telemetry/opentelemetry-go-contrib/plugins/metrics/runtime/internal/metrics"
+)
+
+var (
+	initialisation time.Time
+)
+
+func init() {
+	initialisation = time.Now()
+}
+
+// various runtime metrics have a certain overhead, and as such, we only measure them on a defined interval
+func Runtime(ctx context.Context, meter metric.Meter, interval time.Duration) context.CancelFunc {
+	ctx, cancel := context.WithCancel(ctx)
+
+	ctxTimeout, _ := context.WithTimeout(ctx, time.Second)
```

Comment:
This fragment of code is repeated twice--maybe add the call to `context.WithTimeout` into `metrics.Measure()`? Also, `time.Second` is used twice instead of `interval`, but ideally we'll be able to use Observers for all these metrics, leaving interval up to the metrics SDK.

Reply:
Also, the cancel function should be called at a minimum on the function return. Meaning `defer`ing a call after it is returned would be appropriate.

Reply:
> but ideally we'll be able to use Observers for all these metrics, leaving interval up to the metrics SDK.

@jmacd I am not super familiar with how the observer interval is implemented. If it's tied to the controller push interval, then I think we'd definitely want to decouple the two here. Assuming folks have one meter configured for their service, I think it would be valid for them to set a relatively frequent controller interval (e.g. 1 second), and want to limit the amount of times `runtime.ReadMemStats` to a less frequent interval (e.g. 10 seconds) is called since it is a STW process.